controllers/userController.js

```diff
@@ -0,0 +1,102 @@
+const User = require ('../models/userModel');
+const Student = require ('../models/studentModel');
+const {generateToken} = require("../config/jwtToken");
+const jwt = require("jsonwebtoken");
+const validateMongoDbId = require("../config/validateMongoDbId");
+
+
+//Register a User
+const registerAUser = async (req, res) =>{
+    const email = req.body.email;
+    const findUser = await User.findOne({email: email});
+    if(!findUser) {
+        const newUser = await User.create(req.body)
+        res.status(200).json(newUser);
+    }  
+    else {
+        throw new Error('user already exists')
+    };
+
+}
+
+
+//login a user
+const loginUser = async (req, res) => {
+    const {email, password} = req.body
+
+    const findUser = await User.findOne({email: email});
+    if(findUser && (await findUser.isPasswordMatched(password))) {
+        res.status(200).json({
+            status: true,
```

Comment:
endpoint response should have a schema, should not just create a response on the fly, we can add a response model and create two methods CreateValidLoginResponse, CreateInValidLoginResponse